OMCompiler/Compiler/NFFrontEnd/NFType.mo

```diff
@@ -462,6 +462,17 @@ public
     end match;
   end isComplex;
 
+  function isComplexArray
+    input Type ty;
+    output Boolean isComplex;
+  algorithm
+    isComplex := match ty
+      case ARRAY() guard(isComplex(ty.elementType)) then true;
+      case ARRAY() then isComplexArray(ty.elementType);
```

Comment:
The elementType should never be an array, in that case something has gone very wrong at least as far as the frontend is concerned. So unless the backend is doing something weird with arrays I would suggest removing this case, to avoid confusion about how arrays are represented.

Reply:
okay, i was not sure about this. so the only special case i have to check is Array(Complex) basically?

Reply:
Yes, array types are never nested. More correct would be to use:
```Modelica
case ARRAY() then isComplex(ty.elementType);
case CONDITIONAL_ARRAY() then isComplexArray(ty.trueType);
```
`CONDITIONAL_ARRAY` is only used by the frontend to deal with if-expressions where the two branches have different dimensions, but it's nice to take that into account so the function works correctly for the frontend too.